libsel4vm/src/arch/arm/vgic/vgic.c

```diff
@@ -1146,10 +1135,7 @@ int vm_install_vgic(vm_t *vm)
     vm_memory_reservation_t *vgic_vcpu_reservation = vm_reserve_memory_at(vm, GIC_CPU_PADDR,
                                                                           0x1000, handle_vgic_vcpu_fault, NULL);
     err = vm_map_reservation(vm, vgic_vcpu_reservation, vgic_vcpu_iterator, (void *)vm);
-    if (err) {
-        free(vgic_dist->priv);
-        return -1;
-    }
+    assert(!err);
```

Comment:
We should keep the cleanup and just accept that thing might fail. It's the caller problem then to handle this.